subprojects/junit-jupiter/src/test/java/org/mockitousage/annotation/CaptorAnnotationWithPrimitiveTest.java

```diff
@@ -0,0 +1,52 @@
+package org.mockitousage.annotation;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.doNothing;
+
+@ExtendWith(MockitoExtension.class)
+public class CaptorAnnotationWithPrimitiveTest {
+    @Mock
+    private Foo foo;
+
+    private AutoCloseable closeable;
+
+    static class Foo {
+        void doSomething(int value) {
+        }
+    }
+
+    @BeforeEach
+    void setUp() {
+        closeable = MockitoAnnotations.openMocks(this);
```

Comment:
Nit: the extension should already do this, doesn't it? To ensure we are purely testing the extension, I think we should remove this.

Reply:
You can run/debug the `CaptorAnnotationWithPrimitiveTest` both with and without the changes to see the difference.
Additionally, there is an issue that describes the problem.

Reply:
The problem does not lie in the `openMocks` elsewhere; we will encounter issues with `doNothing().when` before the captor problem.
![Uploading Capture d’écran 2024-02-01 à 17.18.49.png…]()

Reply:
If I understand the user report correctly, I think this reproduces the issue as well:

```java
@ExtendWith(MockitoExtension.class)
public class CaptorAnnotationWithPrimitiveTest {
    @Mock
    private Foo foo;

    static class Foo {
        void doSomething(int value) {
        }
    }

    @Test
    public void shouldCaptorPrimitive(@Captor ArgumentCaptor<Integer> captor) {
        // Given
        int value = 1;

        // When
        doNothing().when(foo).doSomething(captor.capture());

        // Then
        foo.doSomething(value);
        assertEquals(1, captor.getValue());
    }
}
```

Can you update your unit test and confirm that it fails before and succeeds after? If so, please update the regression test to this instead.

Reply:
<img width="753" alt="without_openMocks" src="https://github.com/mockito/mockito/assets/21143371/079e6dd7-3da3-4988-8f8d-bab454cf603a">

Reply:
Using the fix
<img width="529" alt="with_fix" src="https://github.com/mockito/mockito/assets/21143371/87634a1c-9cd9-4725-84d9-f0b484f00326">

Reply:
Thank you!